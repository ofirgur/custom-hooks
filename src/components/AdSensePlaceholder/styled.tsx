import styled from '@emotion/styled';
import { Measurment } from './index';

type AdSensPlaceholderProps = {
  mobile: Record<Measurment, string>
  desktop: Record<Measurment, string>
}
export const AdSensePlaceholderStyled = styled.ins`
  display: block;
  width: ${(props: AdSensPlaceholderProps) => props.mobile.width};
  height: ${(props: AdSensPlaceholderProps) => props.mobile.height};
  margin: 0 auto 40px;
  background: #C4C4C4;

  @media only screen and (min-width: 690px) {
    width: ${(props: AdSensPlaceholderProps) => props.desktop.width};
    height: ${(props: AdSensPlaceholderProps) => props.desktop.height};
  }
`;

type AdvertisementStyled = {
  mobileWidth: string,
  desktopWidth: string
}
export const AdvertisementStyled = styled.div`
  width: ${(props: AdvertisementStyled) => props.mobileWidth};
  text-align: center;
  color: #A1A1A1;
  font-family: Poppins;
  font-size: 11px;
  margin: 40px auto 0;

  @media only screen and (min-width: 690px) {
    width: ${(props: AdvertisementStyled) => props.desktopWidth}
  }
`;
