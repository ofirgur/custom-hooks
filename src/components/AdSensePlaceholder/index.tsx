import React, { useEffect } from 'react';

import { pushToAdsbygoogle } from '../../events';
import { AdSensePlaceholderStyled, AdvertisementStyled } from './styled';

export type Measurment = 'width' | 'height'
type AdSensPlaceholderProps = {
  mobileMeasurments?: Record<Measurment, string>,
  desktopMeasurments?: Record<Measurment, string>
}

const defaultMobile = {
  width: '300px',
  height: '250px',
};

const defaultDesktop = {
  width: '720px',
  height: '90px',
};

const AdSensePlaceholder = ({ mobileMeasurments = defaultMobile, desktopMeasurments = defaultDesktop }: AdSensPlaceholderProps) => {
  useEffect(() => {
    pushToAdsbygoogle();
  }, []);

  return (
    <div>
      <AdvertisementStyled
        mobileWidth={mobileMeasurments.width}
        desktopWidth={desktopMeasurments.width}
      >
        ADVERTISEMENT
      </AdvertisementStyled>
      <AdSensePlaceholderStyled
        className="adsbygoogle"
        data-ad-client="ca-pub-2910351605291781"
        data-ad-slot="8521368831"
        mobile={mobileMeasurments}
        desktop={desktopMeasurments}
      />
    </div>
  );
};

export default AdSensePlaceholder;
