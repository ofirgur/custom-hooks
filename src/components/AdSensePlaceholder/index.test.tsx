import * as React from 'react';
import { render } from '@testing-library/react';
import AdSensePlaceholder from './';

describe('WidgetsDashboard', () => {
  it('should render correctly', () => {
    const { container } = render(<AdSensePlaceholder />);
    expect(container).toMatchSnapshot();
  });
});
