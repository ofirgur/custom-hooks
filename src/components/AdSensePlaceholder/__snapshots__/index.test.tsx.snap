// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WidgetsDashboard should render correctly 1`] = `
<div>
  <div>
    <div
      class="css-1wt90mm"
    >
      ADVERTISEMENT
    </div>
    <ins
      class="adsbygoogle css-1ujbgzu"
      data-ad-client="ca-pub-2910351605291781"
      data-ad-slot="8521368831"
    />
  </div>
</div>
`;
